Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/BreadCrumbs.cpp

```diff
@@ -82,9 +81,38 @@ namespace AzQtComponents
         // clean up the path to use all the first separator in the list of separators
         m_currentPath.replace(g_windowsSeparator, g_separator);
 
+        // update internals
+        m_currentPathSize = m_currentPath.split(g_separator, Qt::SkipEmptyParts).size();
+        m_currentPathIcons.resize(m_currentPathSize);
+
         fillLabel();
     }
 
+    void BreadCrumbs::setDefaultIcon(const QString& icon)
+    {
+        m_defaultIcon = icon;
+    }
+
+    void BreadCrumbs::setIconAt(int index, const QString& icon)
+    {
+        if (index < 0 || index >= m_currentPathSize)
+        {
+            return;
+        }
+
+        m_currentPathIcons[index] = icon;
+    }
+    
+    QIcon BreadCrumbs::iconAt(int index)
```

Comment:
nit: not a big deal but looks like this could be `const` 🙂

Reply:
I saw other functions in this class used int for indices and just followed that, but I agree. We can do a pass and fix it everywhere.